### PR TITLE
[ArrayManager] TST: enable apply tests

### PR DESCRIPTION
**Pull request by jorisvandenbossche**
xref #39146

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
@jreback @jbrockmendel any comments here? (trying to get all testing PRs merged to reduce the custom selection)

File: pandas/tests/apply/test_frame_apply.py [hunk 1]
```diff
@@ -1451,7 +1451,10 @@ def func(row):
     result = df.apply(func, axis=1)
 
     tm.assert_frame_equal(result, expected)
-    tm.assert_frame_equal(df, result)
+    if not using_array_manager:
+        # INFO(ArrayManager) With BlockManager, the row is a view and mutated in place,
+        # with ArrayManager the row is not a view, and thus not mutated in place
```
Comment (jbrockmendel) [inline pandas/tests/apply/test_frame_apply.py:1456]: is there anything we should be asserting in the AM case?

Comment (jorisvandenbossche) [reply pandas/tests/apply/test_frame_apply.py:1456]: Added the equivalent assert (but so asserting it *didn't* change) for AM

**Comment by jorisvandenbossche** [comment]
Failure is an unrelated docstring validation issue that also happens on other PRs

**Pull request merged**